From a94adcb47b9c9058ad9d82fe6833d55e5bb82839 Mon Sep 17 00:00:00 2001
From: xuxiang <xuxiang@eswincomputing.com>
Date: Tue, 14 May 2024 09:58:13 +0800
Subject: [PATCH 028/109] WIN2030-15279:fix:set som pwr last state bug

Changelogs:
1. set som pwr last state when change som power state
2. web-server dip switch callback add dip_switch_soft_ctl

Change-Id: I48a2a1e34fad7ff64399d2c87a6cc90455c9a04d
---
 Core/Src/hf_power_process.c | 4 +---
 Core/Src/web-server.c       | 2 +-
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/Core/Src/hf_power_process.c b/Core/Src/hf_power_process.c
index e9318a2..27d81d4 100644
--- a/Core/Src/hf_power_process.c
+++ b/Core/Src/hf_power_process.c
@@ -109,7 +109,6 @@ void hf_power_task(void *parameter)
 		case STOP_POWER:
 			printf("STOP_POWER\r\n");
 			set_power_off();
-			change_som_power_state(SOM_POWER_OFF);
 			power_state = IDLE_STATE;
 			break;
 		case IDLE_STATE:
@@ -163,7 +162,6 @@ void hf_power_task(void *parameter)
 		case STOP_POWER:
 			printf("STOP_POWER\r\n");
 			set_power_off();
-			change_som_power_state(SOM_POWER_OFF);
 			power_state = IDLE_STATE;
 			break;
 		case IDLE_STATE:
@@ -215,7 +213,6 @@ void set_power_off(void)
 	atx_power_on(pdFALSE);
 	osDelay(10);
 	#endif
-	es_set_som_pwr_last_state(SOM_PWR_LAST_STATE_OFF);
 	pmic_status_led_on(pdFALSE);
 	power_led_on(pdFALSE);
 }
@@ -370,4 +367,5 @@ void change_som_power_state(power_switch_t newState)
 	taskENTER_CRITICAL();
 	som_power_state = newState;
 	taskEXIT_CRITICAL();
+	es_set_som_pwr_last_state(newState);
 }
diff --git a/Core/Src/web-server.c b/Core/Src/web-server.c
index 1ce65e4..87cf51b 100644
--- a/Core/Src/web-server.c
+++ b/Core/Src/web-server.c
@@ -10783,7 +10783,7 @@ int set_dip_switch(DIPSwitchInfo dipSwitchInfo)
 
 	som_dip_switch_state =  ((0x1 & dipSwitchInfo.dip04) << 3) | ((0x1 & dipSwitchInfo.dip03) << 2)
 				| ((0x1 & dipSwitchInfo.dip02) << 1)| (0x1 & dipSwitchInfo.dip01);
-
+	es_set_som_dip_switch_soft_ctl_attr(dipSwitchInfo.swctrl);
 	return es_set_som_dip_switch_soft_state(som_dip_switch_state);
 }
 
-- 
2.25.1

